externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala

```diff
@@ -157,19 +157,20 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
   }
 
   override def closeSession(sessionHandle: SessionHandle): Unit = {
-    if (!userIsolatedSparkSession) {
-      val session = getSession(sessionHandle)
-      if (session != null) {
-        userIsolatedCacheLock.synchronized {
-          if (userIsolatedCacheCount.containsKey(session.user)) {
-            val (count, _) = userIsolatedCacheCount.get(session.user)
-            userIsolatedCacheCount.put(session.user, (count - 1, System.currentTimeMillis()))
+    try {
+      super.closeSession(sessionHandle)
+
+      if (!userIsolatedSparkSession) {
```

Comment:
it reminders me that, shall we only wait for the operations which belong to user ? e.g., for group level sharing, it has no meaning to wait for all operations finish if one user want to close a session.

Reply:
yes, it is useful for CONNECTION share level (spark engine will not be stopped and app could finish even though client connection dies for some reasons). For GROUP share level, no need to wait